zerver/tests/test_message_fetch.py

```diff
@@ -3542,50 +3557,59 @@ def test_invalid_narrow_operand_in_dict(self) -> None:
         # str or int is required for "id", "sender", "channel", "dm-including" and "group-pm-with"
         # operators
         invalid_operands = [["1"], [2], None]
-        error_msg = 'elem["operand"] is not a string or integer'
+        error_msgs = [
+            "operand is not a string or integer",
+            "operand is not a string or integer",
+            "Invalid narrow[0]: Value error, operand is missing",
+        ]
+
         for operand in ["id", "sender", "channel", "dm-including", "group-pm-with"]:
-            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msg)
+            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msgs)
 
         # str or int list is required for "dm" and "pm-with" operator
         invalid_operands = [None]
-        error_msg = 'elem["operand"] is not a string or an integer list'
+        error_msgs = ["Invalid narrow[0]: Value error, operand is missing"]
         for operand in ["dm", "pm-with"]:
-            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msg)
+            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msgs)
 
         invalid_operands = [["2"]]
-        error_msg = 'elem["operand"][0] is not an integer'
+        error_msgs = ["operand[0] is not an integer"]
         for operand in ["dm", "pm-with"]:
-            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msg)
+            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msgs)
 
         # For others only str is acceptable
         invalid_operands = [2, None, [1]]
-        error_msg = 'elem["operand"] is not a string'
+        error_msgs = [
+            "operand is not a string",
+            "Invalid narrow[0]: Value error, operand is missing",
+            "operand is not a string",
+        ]
         for operand in ["is", "near", "has"]:
-            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msg)
+            self.exercise_bad_narrow_operand_using_dict_api(operand, invalid_operands, error_msgs)
 
         # Disallow empty search terms
-        error_msg = 'elem["operand"] cannot be blank.'
+        error_msg = ["operand cannot be blank."]
         self.exercise_bad_narrow_operand_using_dict_api("search", [""], error_msg)
 
     # The exercise_bad_narrow_operand helper method uses legacy tuple format to
     # test bad narrow, this method uses the current dict API format
     def exercise_bad_narrow_operand_using_dict_api(
-        self, operator: str, operands: Sequence[Any], error_msg: str
+        self, operator: str, operands: Sequence[Any], error_msgs: List[str]
     ) -> None:
```

Comment:
This parallel lists API is bad; I think you might just want to define a dataclass or tuple format or something for these operand/error pairs.